zrml/prediction-markets/src/migrations.rs

```diff
@@ -15,7 +15,246 @@
+            || utility::get_on_chain_storage_version_of_market_commons_pallet()
+                != MARKET_COMMONS_NEXT_STORAGE_VERSION
+        {
+            log::info!(
+                "Skipping storage migration for MarketIds; prediction-markets already up to date"
+            );
+            return total_weight;
+        }
+
+        log::info!("Starting storage cleanup of MigrateMarketIdsPerBlockStorage");
+
+        type DisputeBlockToMarketIdsTuple<T> =
+            (<T as frame_system::Config>::BlockNumber, BoundedVec<MarketIdOf<T>, CacheSize>);
+        type IterType<T> = PrefixIterator<DisputeBlockToMarketIdsTuple<T>>;
+
+        let market_ids_per_dispute_iterator: IterType<T> =
+            frame_support::migration::storage_key_iter::<_, _, Twox64Concat>(
+                b"PredictionMarkets",
+                b"MarketIdsPerDisputeBlock",
+            );
+
+        let market_ids_per_dispute: Vec<DisputeBlockToMarketIdsTuple<T>> =
+            market_ids_per_dispute_iterator.collect();
+        total_weight = total_weight
+            .saturating_add(T::DbWeight::get().reads(market_ids_per_dispute.len() as u64));
+        for (dispute_start_block, market_ids) in &market_ids_per_dispute {
+            total_weight = total_weight.saturating_add(T::DbWeight::get().writes(1));
+            let dispute_duration = T::DisputePeriod::get();
+
+            let new_dispute_start_block = dispute_start_block.saturating_add(dispute_duration);
+            MarketIdsPerDisputeBlock::<T>::insert(new_dispute_start_block, market_ids);
+        }
+        for (dispute_start_block, _market_ids) in market_ids_per_dispute {
+            MarketIdsPerDisputeBlock::<T>::remove(dispute_start_block);
+            total_weight = total_weight.saturating_add(T::DbWeight::get().writes(1));
+        }
+
+        let market_ids_per_report_iterator: IterType<T> =
+            frame_support::migration::storage_key_iter::<_, _, Twox64Concat>(
+                b"PredictionMarkets",
+                b"MarketIdsPerReportBlock",
+            );
+
+        let market_ids_per_report: Vec<DisputeBlockToMarketIdsTuple<T>> =
+            market_ids_per_report_iterator.collect();
+        total_weight = total_weight
+            .saturating_add(T::DbWeight::get().reads(market_ids_per_report.len() as u64));
+        for (dispute_start_block, market_ids) in &market_ids_per_report {
+            total_weight = total_weight.saturating_add(T::DbWeight::get().writes(1));
+            let dispute_duration = T::DisputePeriod::get();
+
+            let new_dispute_start_block = dispute_start_block.saturating_add(dispute_duration);
+            MarketIdsPerReportBlock::<T>::insert(new_dispute_start_block, market_ids);
+        }
+        for (dispute_start_block, _market_ids) in market_ids_per_report {
+            MarketIdsPerReportBlock::<T>::remove(dispute_start_block);
+            total_weight = total_weight.saturating_add(T::DbWeight::get().writes(1));
+        }
```

Comment:
Don't you just come in conflict with these removes with what you have written above?

Reply:
https://github.com/zeitgeistpm/zeitgeist/pull/732#discussion_r983274322